src/protagonist/Portal/Features/Batches/Requests/ParseCsv.cs

```diff
@@ -0,0 +1,198 @@
+        customerId = (currentUser.GetCustomerId() ?? -1);
+        maxBatchSize = portalSettings.Value.MaxBatchSize;
+    }
+
+    public async Task<ParseCsvResult> Handle(ParseCsv request, CancellationToken cancellationToken)
+    {
+        var distinctRows = new Dictionary<int, Image>();
+        var readErrors = new List<string>();
+        
+        using (var stream = new StreamReader(request.File.OpenReadStream()))
+        using (var csv = new CsvReader(stream, CultureInfo.InvariantCulture))
+        {
+            csv.Read();
+            csv.ReadHeader();
+            while (csv.Read())
+            {
+                try
+                {
+                    var record = csv.GetRecord<ImageIngestModel>();
+                    if (record.AssetType.ToLower() != "image") continue;
+                    if (distinctRows.ContainsKey(record.Line.Value)) continue;
+                    distinctRows.Add(record.Line.Value, new Image
+                    {
+                        CustomerId = customerId,
+                        ModelId = record.Id,
+                        Space = record.Space,
+                        String1 = record.String1,
+                        String2 = record.String2,
+                        String3 = record.String3,
+                        Origin = record.Origin,
+                        Tags = record.Tags.Split(",",
+                            StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries),
+                        Roles = record.Roles.Split(",",
+                            StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries),
+                        MaxUnauthorised = record.MaxUnauthorized ?? -1,
+                        Number1 = record.Number1,
+                        Number2 = record.Number2,
+                        Number3 = record.Number3,
+                        InitialOrigin = record.InitialOrigin,
+                        Family = AssetFamily.Image,
+                        MediaType = "image/jp2"
+                    });
+                }
+                catch (BadDataException badDataEx)
+                {
+                    return ParseCsvResult.Failure(
+                        $"CSV read error: bad data found in file at line {badDataEx.Context.Parser.Row}, row {badDataEx.Context.Reader.CurrentIndex}");
+                }
+                catch (MissingFieldException missingFieldEx)
+                {
+                    return ParseCsvResult.Failure(
+                        $"CSV read error: line {missingFieldEx.Context.Parser.Row} in file is missing fields");
+                }
+                catch (TypeConverterException typeEx)
+                {
+                    var fieldIndex = typeEx.Context.Reader.CurrentIndex - 1;
+                    var currentLine = csv.GetField(1);
+                    readErrors.Add($"Line {currentLine}: could not parse {ImageIngestModel.FieldNames[fieldIndex]} value ({typeEx.Text})");
```

Comment:
Didn't notice this when I wrote above comment, this _does_ mean `Line` value so I think it's worth using different wording in the other errors.